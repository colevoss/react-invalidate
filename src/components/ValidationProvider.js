// @flow

import { PropTypes, Component, Children } from 'react';
import initializeCoreValidator from '../utils/coreValidator';
import type { CoreValidator } from '../shared/types';
import { CoreValidatorPropTypes } from '../shared/propTypes';

type Props = {
  children: React$Element<*>,
};

// type Context = {
//   validator: CoreValidator,
// };

export default class ValidationProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
  };

  static childContextTypes = {
    validator: CoreValidatorPropTypes,
  };

  constructor(props: Props) {
    super(props);

    this.validator = initializeCoreValidator();
  }

  getChildContext() {
    return {
      validator: this.validator,
    };
  }

  props: Props;
  // context: Context;
  validator: CoreValidator;

  render() {
    return Children.only(this.props.children);
  }
}
