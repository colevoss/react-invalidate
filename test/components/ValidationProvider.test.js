import React from 'react';
import { shallow } from 'enzyme';
import ValidationProvider from '../../src/components/ValidationProvider';

describe('<ValidationProvider />', () => {
  test("renders it child function's component directly", () => {
    const wrapper = shallow(
      <ValidationProvider>
        <div className="test-child" />
      </ValidationProvider>
    );

    expect(wrapper.is('.test-child'));
    expect(wrapper).toMatchSnapshot();
  });

  test('sets this.validator to coreValidator', () => {
    const wrapper = shallow(
      <ValidationProvider>
        <div className="test-child" />
      </ValidationProvider>
    );

    const { validator } = wrapper.instance();

    expect(validator).toMatchSnapshot();
  });
});
