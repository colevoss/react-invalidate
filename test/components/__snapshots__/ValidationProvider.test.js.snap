// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationProvider /> renders it child function's component directly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="test-child"
/>,
  "nodes": Array [
    <div
      className="test-child"
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <ValidationProvider>
        <div
                className="test-child"
        />
</ValidationProvider>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": ValidationProvider {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "children": <div
            className="test-child"
/>,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "validator": Object {
          "areAllValid": [Function],
          "deregisterValidator": [Function],
          "getSubscriptions": [Function],
          "getValidations": [Function],
          "isValidating": [Function],
          "onValidate": [Function],
          "registerValidator": [Function],
          "subscribe": [Function],
          "unsubscribe": [Function],
          "validate": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="test-child"
/>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="test-child"
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ValidationProvider>
    <div
        className="test-child"
    />
</ValidationProvider>,
}
`;

exports[`<ValidationProvider /> sets this.validator to coreValidator 1`] = `
Object {
  "areAllValid": [Function],
  "deregisterValidator": [Function],
  "getSubscriptions": [Function],
  "getValidations": [Function],
  "isValidating": [Function],
  "onValidate": [Function],
  "registerValidator": [Function],
  "subscribe": [Function],
  "unsubscribe": [Function],
  "validate": [Function],
}
`;
